### ISSUE TITLE
Rendering whitespace influences bidi layout

### ISSUE THREAD

**Issue by saeedhei** [microsoft/vscode#99589]
- VSCode Version: 1.45.1
- OS Version: 10 64

I have a json file in my project, emojies is ok in note pad ++ but in vs code is not good UI, 
photo:
[
![jsonUI](https://user-images.githubusercontent.com/25050762/83992919-24106000-a967-11ea-8639-e98181863b9c.png)
](url)

**json:**
```
{
 "invoice_vip":[
    ["🖨️ چاپ فاکتور","🎨 تنظیمات"]
  ]
}
```

**Comment by alexdima** [comment]
I think this is not a complaint about the emojis being rendered colourful, but with the bidi text layout, where the second emoji probably belongs to the second JSON string.

**Comment by saeedhei** [comment]
Yes @alexdima , the output is ok, just in UI, I can't recognize which emoji should I select.

**Comment by NotWearingPants** [comment]
Looks like a duplicate of #83365. The problem in both are the direction-neutral symbols, they both want direction-neutral to be forced LTR. Although this is only useful for structured text (i.e. json), and not for writing an RTL sentence with punctuation.

**Comment by alexdima** [comment]
Our new rendering will be:

<img width="291" alt="image" src="https://user-images.githubusercontent.com/5047891/156365608-a3c69a95-519c-4fcc-894e-5908cd4955e9.png">

**Comment by hediet** [comment]
Selection seems to have issues now:
![recording](https://user-images.githubusercontent.com/2931520/159701148-ca85e7b0-5bc8-4ea4-8ef1-fdb40ea392ac.gif)

**Comment by alexdima** [comment]
Interesting, I think this might be caused by using render whitespace in selection, which influences the rendered line parts.

**Comment by alexdima** [comment]
To verify, configure `"editor.renderWhitespace": "selection"` and open a JSON file with the content:
```json
{
  "invoice_vip":[
    ["🖨️ چاپ فاکتور","🎨 تنظیمات"]
  ]
}
```

Expected: the text doesn't "jump around" when selecting it.

Actual:

https://user-images.githubusercontent.com/5047891/161807955-db85031b-2183-429f-9942-6c141cb3c7a9.mp4


To clarify, the fix will avoid rendering whitespace at all inside RTL tokens. This is not ideal, but better than the jumping the rendering of whitespace causes.

**Comment by Trandung5698** [comment]
Vào 23:58, T.3, 5 Th4, 2022 Alexandru Dima ***@***.***> đã
viết:

**Comment by tinyCoder32** [comment]
Is this gonna be fixed soon? I still see the issue in 1.67
